### PR TITLE
treewide: set GIT_HASH to unknown if `git rev-parse HEAD` fails

### PR DESCRIPTION
**Pull request by NickCao**
## Requirements

- [x] I have read the `CODE_OF_CONDUCT.md`.
- [x] I have updated `CHANGELOG.md` & `tools/install/deb/changelog` to reflect my changes.

## Motivation

Currently building from the release tarball causes a hard failure, as `.git` is not available for embedding the version information.
## Solution
In stead of bailing out, mark the version as unknown.

### CODE REVIEW

**Comment by ltabis** [comment]
Good catch @NickCao, thank you ! (I just modified your PR a bit by adding the PR number to the changelog)

**Review by ltabis — APPROVED** [review]

File: src/vsmtp/vsmtp-core/build.rs [hunk 1]
```diff
@@ -21,11 +21,11 @@ fn main() {
     let output = Command::new("git")
         .args(["rev-parse", "HEAD"])
         .output()
-        .expect("failed to get git commit hash");
+        .map(|out| out.stdout).unwrap_or("unknown".as_bytes().to_vec());
```
Comment (ltabis) [inline src/vsmtp/vsmtp-core/build.rs:24]: Could you please format this line properly ?

```rust
-        .map(|out| out.stdout).unwrap_or("unknown".as_bytes().to_vec());
+        .map(|out| out.stdout)
+        .unwrap_or("unknown".as_bytes().to_vec());
```

**Review by ltabis — CHANGES_REQUESTED** [review]
There are formatting errors emitted by rustfmt, please fix them :)

File: src/vsmtp/vsmtp-core/build.rs (hunk 1 above)
Comment (Mathieu-Lala) [reply src/vsmtp/vsmtp-core/build.rs:24]: I will do it in another PR, good to go!

**Pull request merged**